UnitTests/GitUITests/CommandsDialogs/FormBrowseTests.LeftPanel.ReorderNodes.cs

```diff
@@ -174,12 +171,22 @@ private void ValidateOrder(List<TreeNode> initialNodes, TreeNodeCollection currN
             }
         }
 
-        private void RunFormTest(Action<FormBrowse> testDriver)
+        private void RunFormTest(Action<RepoObjectsTree.TestAccessor> testDriver)
         {
             RunFormTest(
                 form =>
                 {
-                    testDriver(form);
+                    var repoObjectTree = form.GetTestAccessor().RepoObjectsTree.GetTestAccessor();
+                    var currNodes = repoObjectTree.TreeView.Nodes;
+
+                    // wait up to 5 seconds until the nodes will have been loaded
+                    for (int i = 0; i < 50 && currNodes.Count != 4; ++i)
+                    {
+                        Thread.Sleep(100);
+                        Application.DoEvents();
+                    }
```

Comment:
🤔

Reply:
This is safer than `ThreadHelper.JoinPendingOperationsAsync` which might block forever.
I was not sure when async actions get their run. Perhaps `Application.DoEvents` can be omitted.

Reply:
`ThreadHelper.JoinPendingOperationsAsync` is a fundamental correctness requirement.

Reply:
😕 Can you clarify where `JoinPendingOperationsAsync` was being used?

Reply:
`JoinPendingOperationsAsync` is referenced by:
- `AsyncTestHelper.WaitForPendingOperations`
- `AfterTest`
- `FormCommitTests`, which should use `AsyncTestHelper.WaitForPendingOperations` instead of `await` (I'll change this.)
- `ThreadHelperTests`

Reply:
> ensuring ongoing work does not interfere with downstream tests

I agree, though the caveat of both `ThreadHelper.JoinPendingOperationsAsync` and its wrapper `AsyncTestHelper.WaitForPendingOperations` is that they might block forever. The NUnit test runner often fails to detect and to enforce timeouts on an hanging test case.

Reply:
It would be fine to add a timeout which caused the test to fail if it did not complete within a maximum time. Future calls to initialize `ThreadHelper` should throw an exception that NUnit recognizes as a skipped test.

Reply:
I've activated the per-test-case timeout of the NUnit test runner. It works for long running tests.
But unfortunately, it is unable to cancel (most of) our hanging tests.